src/compute/arity_assign.rs

```diff
@@ -48,54 +58,39 @@ where
     // bitmap has an offset.
     if let Some(rhs) = rhs.validity() {
         if lhs.validity().is_none() {
-            *lhs = lhs.with_validity(Some(rhs.clone()))
+            lhs.set_validity(Some(rhs.clone()));
         } else {
             lhs.apply_validity(|bitmap| {
-                // we need to take ownership for the `into_mut` call, but leave the `&mut` lhs intact
-                // so that we can later assign the result to out `&mut bitmap`
-                let owned_lhs = std::mem::take(bitmap);
-
-                *bitmap = match owned_lhs.into_mut() {
-                    // we take alloc and write to new buffer
+                match bitmap.into_mut() {
                     Either::Left(immutable) => {
-                        // we allocate a new bitmap
+                        // alloc new region
                         &immutable & rhs
                     }
-                    // we can mutate in place, happy days.
                     Either::Right(mut mutable) => {
+                        // mutate in place
                         let mut mutable_ref = &mut mutable;
                         mutable_ref &= rhs;
                         mutable.into()
                     }
                 }
             });
         }
-    }
+    };
 
-    // we need to take ownership for the `into_mut` call, but leave the `&mut` lhs intact
-    // so that we can later assign the result to out `&mut lhs`
-    let owned_lhs = std::mem::take(lhs);
-
-    *lhs = match owned_lhs.into_mut() {
```

Comment:
this is the important change - this was trying to get a `MutableBitmap`, but we do not need one - we only need a mut reference to the values.

Reply:
Yep, good catch.